### PR TITLE
Add BigQueryError field to BigQueryException and tests

### LINKED ISSUE #444: Tie BigQueryError to BigQueryException
`BigQueryError`s are returned as part of a job state, as the result of a [query operation](https://cloud.google.com/bigquery/docs/reference/v2/jobs/query) and as a response to [insert all](https://cloud.google.com/bigquery/docs/reference/v2/tabledata/insertAll). In all the above cases a `BigQueryException` is not thrown.

When a request fails the service returns an error in the form:
```json
{
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "notFound",
    "message": "Not found: Table gcloud-devel:dataset.view"
   }
  ],
  "code": 404,
  "message": "Not found: Table gcloud-devel:dataset.view"
 }
}
```
So we could add a `BigQueryError` field to `BigQueryException` containing the error that caused the exception.

### PR DESCRIPTION
**Pull request by mziccard**
This PR ties BigQueryError to BigQueryException and adds some integration tests (fixes #444).

### CODE REVIEW

File: gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQueryException.java [hunk 1]
```diff
@@ -31,10 +31,25 @@
   private static final long serialVersionUID = -5504832700512784654L;
   public static final int UNKNOWN_CODE = -1;
 
+  private BigQueryError error;
```
Comment (ajkannan) [inline gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQueryException.java]: nit: could this be made final?

File: gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/DefaultBigQueryRpc.java [hunk 2]
```diff
@@ -91,7 +92,14 @@ private static BigQueryException translate(IOException exception) {
 
   private static BigQueryException translate(GoogleJsonError exception) {
     boolean retryable = RETRYABLE_CODES.contains(exception.getCode());
-    return new BigQueryException(exception.getCode(), exception.getMessage(), retryable);
+    BigQueryError bigqueryError = null;
+    if (exception.getErrors() != null  && !exception.getErrors().isEmpty()) {
+      GoogleJsonError.ErrorInfo error = exception.getErrors().get(0);
```
Comment (ajkannan) [inline gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/DefaultBigQueryRpc.java]: Do you know whether it's possible to get multiple errors in the same `GoogleJsonError` (and if so, would it be helpful to return a list of errors)?  I realize I also only [take the first error](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-resourcemanager/src/main/java/com/google/gcloud/spi/DefaultResourceManagerRpc.java#L60) in Resource Manager as well

**Comment by ajkannan** [comment]
Looks good, one nit and one question from me.

File: gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/DefaultBigQueryRpc.java (hunk 2 above)
Comment (mziccard) [inline gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/DefaultBigQueryRpc.java]: And if I am not mistaken this is also the case in Datastore.
But no, I don't know whether there are some cases in which the `errors` array contains more than one value. Maybe @aozarov has more information than I do?

Comment (aozarov) [inline gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/DefaultBigQueryRpc.java]: I think the api client had it as a list for flexibility but so far I am not aware of services using that and prefered
to keep the API simple in that case.

I think It should not be hard to nest the Exceptions later on if needed.

**Comment by aozarov** [comment]
Looks good to me too (after fixing the nit @ajkannan mentioned).

**Comment by mziccard** [comment]
Fixed comment in place!

**Pull request merged**